### PR TITLE
(add) New project

### PR DESCRIPTION
**Pull request by Gautam-Arora24**
Added new project.

Signed-off-by: Gautam Arora <gautamarora6248@gmail.com>

### CODE REVIEW

File: projects/Gautam-Arora24/resize-observer-hook.mdx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    url: https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserverEntry
+---
+
+<!-- Divide your content into sections using our handy shortcodes. Each section is optional -->
+
+<Overview>
+"resize-observer-hook" is a ⚛️ React hook that monitors an element (provided with ref) and provides the height and width of that element dynamically using Native Resize Observer API.
+<br/>
+<br/>
+We usually use media queries to change the style of a particular element but media queries are great for updating layouts at specific points when the viewport changes sizes, but what if we want to change the style of an element based on the size of another element. We can use this library for the same case.
```
Comment (NJalal7) [inline projects/Gautam-Arora24/resize-observer-hook.mdx:36]:
```suggestion
We usually use media queries to change the style of a particular element as media queries are great for updating layouts at specific points when the viewport changes sizes. But what if we want to change the style of an element based on the size of another element? In that case, we can use this library!
```

File: projects/Gautam-Arora24/resize-observer-hook.mdx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+Currently, the library is in the initial stage, and there are a lot more things that need to be added like Types support, Testing for different cases, more props and etc. So, if you are interested to contribute to the library, you are welcomed 🙂.
```
Comment (NJalal7) [inline projects/Gautam-Arora24/resize-observer-hook.mdx:37]:
```suggestion
Currently, the library is in the initial stage, and there are a lot more things that need to be added like Types support, testing for different cases, more props, etc. So, if you are interested to contribute to the library, you are welcomed 🙂.
```

File: projects/Gautam-Arora24/resize-observer-hook.mdx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+<br/>Do ⭐️ the repo.
+<br/>
+Any feedback or suggestions are appreciated 👋
+
+</Overview>
+
+<Contributing>
+
+- If you feel like there is some kind of bug or any other issue that is concerning for the library. Feel free to create an issue. I will try my best to solve it as soon as possible.
```
Comment (NJalal7) [inline projects/Gautam-Arora24/resize-observer-hook.mdx:46]:
```suggestion
- If you feel that there may be some kind of bug or any other issue that is concerning for the library, feel free to create an issue. I will try my best to solve it as soon as possible.
```

File: projects/Gautam-Arora24/resize-observer-hook.mdx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+name: resize-observer-hook.
+repoUrl: http://github.com/Gautam-Arora24/resize-observer-hook
+description: ⚛️ A React Hook to monitor changes in the size of an element using native ResizeObserver API 🔍
+languages: # A list of programming languages used in this project
+  - JavaScript
+  - HTML / CSS
+  - React
+tags:
+  - First Timer Friendly
+  - Resize-Observer
```
Comment (NJalal7) [inline projects/Gautam-Arora24/resize-observer-hook.mdx:11]:
```suggestion
```

File: projects/Gautam-Arora24/resize-observer-hook.mdx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+---
+name: resize-observer-hook.
```
Comment (NJalal7) [inline projects/Gautam-Arora24/resize-observer-hook.mdx:2]:
```suggestion
name: resize-observer-hook
```

**Review by NJalal7 — APPROVED** [review]
Thanks for adding resize-observer-hook to OSS Port! We're excited to help you find contributors. Should you be interested in adding a CodeSee Map to help folks onboard to the codebase, feel free to submit another PR!

**Comment by Gautam-Arora24** [comment]
Thank you so much for merging the PR! I'll create a PR for CodeSee Map soon :)

**Pull request merged**